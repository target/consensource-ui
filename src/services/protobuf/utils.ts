import {
  CertificateRegistryPayload,
  ICertificateRegistryPayload,
} from './compiled';

export interface PayloadInfo {
  payloadBytes: string | Buffer | NodeJS.TypedArray | DataView;
  inputs: string[];
  outputs: string[];
}

export const ACTIONS = CertificateRegistryPayload.Action;

/**
 * Encodes a CertificateRegistryPayload message
 * @param message CertificateRegistryPayload message or plain object to encode
 */
export function encodePayload(
  payload: ICertificateRegistryPayload,
): Uint8Array {
  return CertificateRegistryPayload.encode(payload).finish();
}
