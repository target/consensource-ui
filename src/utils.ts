// Util collection

/**
 * Selects keys from an object
 */
export const pluck = (obj: any, ...keys: Array<any>): any =>
    keys.reduce((o, k) => {
        if (typeof obj[k] !== 'undefined') {
            o[k] = obj[k];
        }
        return o;
    }, {});

/**
 * Merges objects, starting from left to right
 */
