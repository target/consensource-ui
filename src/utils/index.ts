export function getLocaleFromUnix(timestampSeconds: number) {
  const timestampMs = timestampSeconds * 1000;
  return new Date(timestampMs).toLocaleDateString();
}
