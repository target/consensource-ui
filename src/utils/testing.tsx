import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { StoresContext } from 'view/context';
import { stores } from 'stores';
import CssBaseline from '@material-ui/core/CssBaseline';
import { TestThemeProvider } from './testThemeProvider';

const wrapper = ({ children }: any) => {
  return (
    <CssBaseline>
      <StoresContext.Provider value={stores}>
        <TestThemeProvider>
          <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
        </TestThemeProvider>
      </StoresContext.Provider>
    </CssBaseline>
  );
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
