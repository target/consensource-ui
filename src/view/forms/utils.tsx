import React from 'react';
import { Grid, Typography, TextFieldProps } from '@material-ui/core';
import { BatchStatusRes } from 'services/api';

export interface TransactionFormProps {
  setBatchStatusLink: (statusLink: BatchStatusRes['link']) => void;
}

export type onChangeEvent = Parameters<
  NonNullable<TextFieldProps['onChange']>
>[0];

/**
 * Helper function that checks if any values in `state` have not been set.
 * Returns true if there is any key in `state` that is falsey.
 */
export function hasEmptyFields<T>(state: T) {
  return Object.keys(state).some((key) => !state[key as keyof T]);
}

export interface FormErrMsgProps {
  msg?: string;
}

export function FormErrMsg({ msg }: FormErrMsgProps) {
  if (!msg) {
    return null;
  }

  return (
    <Grid item>
      <Typography variant="caption" color="error">
        {msg}
      </Typography>
    </Grid>
  );
}
