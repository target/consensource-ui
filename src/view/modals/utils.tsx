import { DialogProps as MUIDialogProps } from '@material-ui/core';
import { FactoryResData } from 'services/api';

export interface DialogProps extends MUIDialogProps {
  handleClose: () => void;
}

export interface OrgDialogProps extends DialogProps {
  existing_org: FactoryResData;
}
