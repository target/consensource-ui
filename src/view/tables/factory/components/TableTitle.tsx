import React from 'react';
import { Typography, Grid, makeStyles, createStyles } from '@material-ui/core';
import { Business as BusinessIcon } from '@material-ui/icons';

const useStyles = makeStyles(
  createStyles({
    icon: {
      paddingTop: 10,
    },
  }),
);

export const TableTitle = () => {
  const classes = useStyles();

  return (
    <Grid container alignItems="center" spacing={2}>
      <Grid item>
        <Typography variant="h4">Factories</Typography>
      </Grid>
      <Grid item>
        <BusinessIcon fontSize="large" className={classes.icon} />
      </Grid>
    </Grid>
  );
};
