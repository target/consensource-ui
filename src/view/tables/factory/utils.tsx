import { MUIDataTableOptions, MUIDataTableChip } from 'mui-datatables';

export const DEFAULT_ROWS_PER_PAGE = 100;

export const textLabels: MUIDataTableOptions['textLabels'] = {
  body: {
    noMatch: 'No factories found',
  },
};

/**
 * Default props for the filter chips displayed in the
 * `<TableFilterList />`.
 */
export const filterChipProps: MUIDataTableChip = {
  color: 'secondary',
  variant: 'default',
};

/**
 * Utility function to convert a string to an array, or return
 * the original param if an array was passed.
 *
 * This operation is frequent when decoding URL query params.
 */
export const convertStrToArray = (val: string[] | string) => {
  return typeof val === 'string' ? [val] : val;
};
