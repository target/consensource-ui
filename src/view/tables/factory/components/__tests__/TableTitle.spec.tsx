import React from 'react';
import { render } from 'utils/testing';
import { TableTitle } from '../TableTitle';

describe('<TableTitle />', () => {
  it('renders', () => {
    const { container } = render(<TableTitle />);
    expect(container).toMatchSnapshot();
  });
});
