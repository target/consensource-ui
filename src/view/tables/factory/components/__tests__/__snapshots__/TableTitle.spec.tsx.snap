// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableTitle /> renders 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Factories
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon MuiSvgIcon-fontSizeLarge"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
        />
      </svg>
    </div>
  </div>
</div>
`;
